src/main/java/edu/wpi/grip/core/Step.java

```diff
@@ -53,21 +46,21 @@ public Operation getOperation() {
     /**
      * @return An array of <code>Socket</code>s that hold the inputs to this step
      */
-    public Socket<?>[] getInputSockets() {
+    public InputSocket<?>[] getInputSockets() {
         return inputSockets;
     }
 
     /**
      * @return An array of <code>Socket</code>s that hold the outputs of this step
      */
-    public Socket<?>[] getOutputSockets() {
+    public OutputSocket<?>[] getOutputSockets() {
         return outputSockets;
     }
 
     @Subscribe
     public void onInputSocketChanged(SocketChangedEvent e) {
         // If this socket that changed is one of the inputs to this step, run the operation with the new value.
-        if (e.getSocket().getStep() == this) {
+        if (Arrays.asList(inputSockets).contains(e.getSocket())) {
```

Comment:
Revert this change.  This has absolutely nothing to do with the rest of the commit.